Tensor.lua

```diff
@@ -5,14 +5,14 @@ local Storage = {}
 local Tensor = {}
 
 -- types
-local types = {'Byte', 'Char', 'Short', 'Int', 'Long', 'Float', 'Double'}
+local types = {'Byte', 'Char', 'Short', 'Int', 'Long', 'Float', 'Half', 'Double'}
```

Comment:
@soumith, @gchanan: I think I have found the right way to handle it: please check out the last commit. 
Also, I have updated https://github.com/torch/cutorch/pull/578 - see how the CUDA  'half' is passed to TH